packages/itwinui-react/src/core/ButtonGroup/ButtonGroup.tsx

```diff
@@ -172,6 +176,12 @@ const BaseGroup = React.forwardRef((props, forwardedRef) => {
 
 // ----------------------------------------------------------------------------
 
+type OverflowGroupProps = Pick<
+  ButtonGroupProps,
+  'children' | 'orientation' | 'overflowButton' | 'overflowPlacement'
+>;
+
+/** Note: If `overflowButton == null`, it behaves like a `BaseGroup`. */
```

Comment:
This component doesn't need to handle `overflowButton == null` because it's already taken care of in the parent component. We can just assume that `overflowButton` will not be null; the types can be changed to enforce this.